crypto/src/conversation.rs

```diff
@@ -329,11 +334,26 @@ impl MlsConversation {
         let message = match message {
             ProcessedMessage::ApplicationMessage(app_msg) => (Some(app_msg.into_bytes()), None),
             ProcessedMessage::ProposalMessage(proposal) => {
-                self.group.store_pending_proposal(*proposal);
+                if is_external_add_proposal(proposal.as_ref()) {
+                    let other_clients = self
+                        .group
+                        .members()
+                        .iter()
+                        .map(|kp| kp.credential().identity())
+                        .collect();
+                    let add_proposal = as_add_proposal(proposal.proposal())?;
+                    let callbacks = callbacks.as_ref().ok_or(CryptoError::CallbacksNotSet)?;
+                    if !callbacks
+                        .is_external_proposal_valid(add_proposal.key_package().credential().identity(), other_clients)
+                    {
+                        return Err(CryptoError::ExternalProposalError);
```

Comment:
error could be a bit more contextualized here e.g. "only external add proposals for users already in group are allowed"